sdk/search/Azure.Search.Documents/tests/Samples/Sample01_HelloWorld.cs

```diff
@@ -49,6 +49,8 @@ public async Task CreateClient()
             Response<SearchServiceStatistics> stats = indexClient.GetServiceStatistics();
             Console.WriteLine($"You are using {stats.Value.Counters.IndexCounter.Usage} indexes.");
             #endregion Snippet:Azure_Search_Tests_Samples_CreateClient
+
+            Assert.GreaterOrEqual(stats.Value.Counters.IndexCounter.Usage, 1);
```

Comment:
Is this timing issue? Curious how this resulted in tests failure when run all together vs not!

Reply:
No, this is a different test and yes, it's timing issue for this test case.

Previously [Options](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2109985&view=logs&j=32e69777-6632-566b-8772-98de52ef067a&t=d8987eda-cbe3-5fa2-c061-f64e97907b47&l=528) test case was failing. The issue was the common [EnsureInitialized](https://github.com/ShivangiReja/azure-sdk-for-net/blob/52ab5e3dd1a9b553f6aea85b74c4d6b6ea068dc7/sdk/search/Azure.Search.Documents/tests/Utilities/SharedSearchResources.cs#L31) method, which we were using for sample and test both. Now, we use TestDocumentsForSample(uploads document in pascal case like public portal) for samples and TestDocument(upload document in camel case) for tests (see [here](https://github.com/ShivangiReja/azure-sdk-for-net/blob/52ab5e3dd1a9b553f6aea85b74c4d6b6ea068dc7/sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs#L424-L431)). [EnsureInitialized](https://github.com/ShivangiReja/azure-sdk-for-net/blob/52ab5e3dd1a9b553f6aea85b74c4d6b6ea068dc7/sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs#L273) method was initializing for tests and using the same test resources for both tests and samples, so some of samples were failing. [Here](https://github.com/Azure/azure-sdk-for-net/pull/33364/files#diff-b9e83f9dcb19a960cc355b5ed62e8b3337d10446f1d86f41ce4fb3fc96d55a3c) I added the fix for it. 

Now all the test cases are passing and all the pipelines are green -- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2115015&view=results.